3rdparty/mshadow/mshadow/base.h

```diff
@@ -988,6 +1034,7 @@ struct minimum {
 };
 }  // namespace red
 
+#ifndef __NVCC__
```

Comment:
I don't like it - can we make a similar thing as in the case of fp16 for CPUs that did not support F16C instructions (i.e. code that runs but may be slower than the code for hardware natively supporting bfloat16)?

Reply:
You can implement `atomicAdd` (which seems to be the problem you are facing)with `atomicCAS` like this: https://github.com/apache/incubator-mxnet/blob/master/src/common/cuda_utils.h#L702-L721

Reply:
We don't have enough background / knowledge to enable Bfloat16 on GPU side. So probably we can't make the change you proposed. Alternately, any code refactoring on GPU side is welcome. you may change this in following PR.